lib/puppet/functions/extlib/debug/break.rb

```diff
@@ -0,0 +1,108 @@
+begin
+  require 'puppet-debugger'
+rescue LoadError => e
+  Puppet.err('You must install the puppet-debugger: gem install puppet-debugger')
+end
+
+# @summary Injects a puppet debugger session where the break point was placed.
+#
+# This function allows you to set a breakpoint and break into the puppet compiler
+# allowing you to interact with the puppet manifest.  Think of this like a 
+# interactive manifest where the scope, facts and code is setup for you to play
+# with in real time. This function is similar to binding.pry for ruby but instead
+# is use solely  with the puppet debugger which is just for puppet code only.
+#
+# ```ruby
+# Ruby Version: 2.5.1
+# Puppet Version: 6.4.0
+# Puppet Debugger Version: 0.10.3
+# Created by: NWOps <corey@nwops.io>
+# Type "commands" for a list of debugger commands
+# or "help" to show the help screen.
+#
+#         8:   service{'httpd': ensure => running}
+#         9:    $var2 = ['value1', 'value2'] 
+#         10:   # how to find values with an empheral scope
+#         11:   $var2.each | String $item | {
+#         12:     file{"/tmp/${item}": ensure => present}
+#      => 13:     extlib::debug::break()
+#         14:   }
+#         15:   extlib::debug::break()
+#         16:   if $var1 == 'value1' {
+#         17:     extlib::debug::break()
+#         18:   }
+# 1:>> $item
+# => "value1"
+# ````
+#
+# @example How to find values with an empheral scope.
+#  $var2.each | String $item | {
+#    file{"/tmp/${item}": ensure => present}
+#    extlib::debug::break()
+#  }
+#
+# @example - Insert a breakpoint anywhere in your puppet code.
+#   extlib::debug::break()
+#
+#
+Puppet::Functions.create_function(:'extlib::debug::break', Puppet::Functions::InternalFunction) do
+  # the function below is called by puppet and and must match
+  # the name of the puppet function above.
+
+  # The debugger_stack_count is used to determine if this function has already been called
+  # in order to prevent additionally help output from the debugger 
+
+  def initialize(scope, loader)
+    super
+    @debugger_stack_count = 0
+  end
+
+  # @return [Integer] - returns the nunber of debugger instances currently alive
+  # @param options [Hash] - a hash of puppet debugger options, not required
```

Comment:
Specify `[Hash]` here should be redundant.  I believe for ruby functions (vs puppet language functions), this is also true for `@return`.